crates/api/src/local_user/save_settings.rs

```diff
@@ -120,15 +120,15 @@ impl Perform for SaveUserSettings {
     .map_err(|e| LemmyError::from_error_message(e, "user_already_exists"))?;
 
     if let Some(discussion_languages) = data.discussion_languages.clone() {
-      let discussion_languages: Vec<LocalUserLanguageForm> = discussion_languages
-        .into_iter()
-        .map(|lang| LocalUserLanguageForm {
-          local_user_id,
-          language_id: lang,
-        })
-        .collect();
+      // An empty array is a "clear" / set all languages
```

Comment:
We'll need to remember this when doing the front end of possible language selections. IE deselecting all, or an "all languages" button could send an empty vector.

Reply:
Sending all languages should also work, so this would be more like an optimization, and probably not necessary for a relatively rare operation like this.

Reply:
Its good to have a distinction between API saves that are: "don't update" this field, and "clear this field" for the optionals. 

Its kinda similar to the two types of Nulls, that diesel handles as `None` or `Some(None)`.